widget/widget_renderer.go

```diff
@@ -0,0 +1,34 @@
+package widget
+
+import (
+	"fyne.io/fyne"
+	"fyne.io/fyne/theme"
+)
+
+// When using the baseRenderer the embedding renderer should call
+// layoutShadow(contentSize, contentPos) to lay out the shadow.
+type baseRenderer struct {
+	objects []fyne.CanvasObject
+	sh      fyne.CanvasObject
+}
+
+func newBaseRenderer(objects []fyne.CanvasObject, level int) *baseRenderer {
+	var shadow fyne.CanvasObject
+	if level > 0 {
+		shadow = newShadow(shadowAround, level*theme.Padding()/2)
```

Comment:
It isn't immediately obvious what level means here. Can this be 'depth' to match newShadow and push this calculation out to the callers?

Reply:
Also, can depth be updated after the renderer is created? Eg. the theme gets updated and now theme.Padding() is different

Reply:
I thought of it as the _level_ of elevation (https://material.io/design/environment/elevation.html) which is then used to compute the _depth_ of the shadow.
The depth depends on the theme while the level doesn't.
I don't want to use the same name for both but I'm open for alternatives.

Reply:
Regarding the update: Probably the whole theme dependency should be moved into the shadow which will be refreshed on theme change and can adjust its size, then.
WDYT?

Reply:
How about calling it elevation?

So this would change newShadow to accept elevation and it will calculate the depth using the current theme and respond to theme changes?

Reply:
Elevation seems to be the name that Material uses, but it seems a fairly long name https://material.io/design/environment/light-shadows.html#

Reply:
That's why I used `level` … it shares 80% of its letters with `elevation` :).